crates/rolldown/src/stages/generate_stage/code_splitting.rs

```diff
@@ -740,11 +740,8 @@ impl GenerateStage<'_> {
     let mut pending_common_chunks: FxIndexMap<BitSet, Vec<ModuleIdx>> = FxIndexMap::default();
     // If it is allow to allow that entry chunks have the different exports as the underlying entry module.
     // This is used to generate less chunks when possible.
-    let allow_optimize_chunk = (!matches!(self.options.preserve_entry_signatures, PreserveEntrySignatures::Strict)
-        || !self.link_output.overrode_preserve_entry_signature_map.is_empty())
-          // partial workaround of https://github.com/rolldown/rolldown/issues/5026#issuecomment-2990146735
-          // TODO: maybe we could bailout peer chunk?
-        && !self.link_output.metas.iter().any(|meta| meta.is_tla_or_contains_tla_dependency);
+    let allow_optimize_chunk =
```

Comment:
```suggestion
    // partial workaround of https://github.com/rolldown/rolldown/issues/5026#issuecomment-2990146735
    // TODO: maybe we could bailout peer chunk?
    let allow_optimize_chunk =
```
We should keep this comment.

Reply:
fixed